python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -1101,6 +1111,33 @@ def _find_stars(self, exposure, background, id_generator, background_to_photomet
         sources = detections.sources
         self.star_sky_sources.run(exposure.mask, id_generator.catalog_id, sources)
 
+        n_sky_sources = np.sum(sources["sky_source"])
+        if (self.config.do_downsample_footprints
+           and (len(sources) - n_sky_sources) > self.config.downsample_max_footprints):
+            if exposure.info.id is None:
+                self.log.warning("Exposure does not have a proper id; using 0 seed for downsample.")
+                seed = 0
+            else:
+                seed = exposure.info.id & 0xFFFFFFFF
+
+            gen = np.random.RandomState(seed)
+
+            # Isolate the sky sources before downsampling.
+            indices = np.arange(len(sources))[~sources["sky_source"]]
+            indices = gen.choice(
+                indices,
+                size=self.config.downsample_max_footprints,
+                replace=False,
+            )
+            skyIndices, = np.where(sources["sky_source"])
+            indices = np.concatenate((indices, skyIndices))
+
+            self.log.info("Downsampling from %d to %d footprints.", len(sources), len(indices))
```

Comment:
I think in the log message, the code and the config docstring it'd be nice to just say in words what the intention is wrt the sky sources so that future maintainers know what _should_ be being done, and therefore just add the sky source count to the log message so it's clear.